app/javascript/components/mentoring/discussion/DiscussionPostList.tsx

```diff
@@ -1,75 +1,219 @@
       channel.disconnect()
     }
-  }, [discussionId, refetch])
+  }, [cacheKey, discussionId])
 
-  if (isLoading) {
+  if (status === 'loading') {
     return (
       <div role="status" aria-label="Discussion post list loading indicator">
         <Loading />
       </div>
     )
   }
 
-  if (isSuccess && data) {
+  if (data) {
     return (
       <div className="discussion">
-        {iterations.map((iteration) => {
+        {hasNewMessages ? (
+          <button type="button" onClick={() => scrollToLastMessage()}>
+            View new messages
+          </button>
+        ) : null}
+        {iterationsWithPosts.map((iteration) => {
           return (
-            <div key={iteration.idx}>
-              <div>Iteration {iteration.idx}</div>
+            <React.Fragment key={iteration.idx}>
+              <IterationMarker
+                idx={iteration.idx}
+                createdAt={iteration.createdAt}
+              />
+              <details className="c-details auto-feedback">
+                <summary>
+                  <GraphicalIcon icon="alert-circle" className="info-icon" />
+                  <div className="info">
+                    WDNewmann recieved automated feedback
+                  </div>
+                  <GraphicalIcon
+                    icon="chevron-right"
+                    className="--closed-icon"
+                  />
+                  <GraphicalIcon icon="chevron-down" className="--open-icon" />
+                </summary>
+                <div className="feedback">
+                  <div className="c-textual-content --small">
+                    <p>
+                      The biggest improvement you could make to this would be to
+                      use Ruby's String helper methods rather than regular
+                      expressions. That will help make it much more readable to
+                      future-you or another developer. Take a look at the String
+                      docs and check out things such as <code>strip</code> and{' '}
+                      <code>end_with?</code>.
+                    </p>
+                  </div>
+                  <div className="byline">
+                    <Avatar
+                      src="https://avatars2.githubusercontent.com/u/5337876?s=460&v=4"
+                      handle="mentor handle"
+                    />
+                    <div className="name">by ErikSchierboom</div>
+                    {/* TODO: Extract to common reputation component */}
+                    <div className="c-reputation" aria-label="20 reputation">
+                      <GraphicalIcon icon="reputation" />
+                      <span>20</span>
+                    </div>
+                  </div>
+                  <a href="#" className="more">
+                    Learn more about this feedback
+                    <Icon icon="external-link" alt="Opens in new tab" />
+                  </a>
+                </div>
+              </details>
```

Comment:
@kntsoriano This needs implementing. For each iteration there may (or may not) be a piece of automated feedback. If there is, it should render in this block.